packages/common/src/lib/color-picker-form-field/color-picker-form-field.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, EventEmitter, forwardRef, Output } from '@angular/core';
+import { Input } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import tinycolor from "tinycolor2";
+
+@Component({
+  selector: 'igo-color-picker-form-field',
+  templateUrl: './color-picker-form-field.component.html',
+  styleUrls: ['./color-picker-form-field.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => ColorPickerFormFieldComponent),
+      multi: true,
+    }
+  ]
+})
+export class ColorPickerFormFieldComponent implements ControlValueAccessor {
+
+  // native color picker can't give transparency
+  // set transparency manually
+  @Input() setAlpha: number = 1;
+
+  // by default native color picker gives hex result
+  @Input() outputFormat: 'hex'| 'rgba'| 'hsla' = 'rgba';
+
+  // default color
+  @Input() color: string = '#000';
+
+  // open and close native color picker event
+  @Output() open = new EventEmitter<boolean>();
+  @Output() close = new EventEmitter<boolean>();
+
+  onChange: any = () => {};
+  onTouch: any = () => {};
+
+  // final value we get as result with custom format
+  set value(value: string) {
+    value = (!value) ? this.color : value ;
+    this.onChange(value);
+    this.onTouch(value);
+    this._value = value;
+  }
+  get value (): string {
+    return this._value;
+  }
+  public _value: string;
+
+  // native color picker accept hex format
+  set hexColor(value: string) {
+    this.value = this.setFormat(value);
+    this._hexColor = value;
+  }
+  get hexColor (): string {
+    if(this.color) {
+      this._hexColor = tinycolor(this.color).setAlpha(this.setAlpha).toHexString();
```

Comment:
Le hexColor est attaché a un NgModel donc il est dynamique mais dans le getter, on peut overrider la valeur par le `@Input color`? Ça pourrait causer un drole de comportement?

Idéalement si on veut gérer les inputs et les attribuer en tant que valeur par défaut, on utilise les lifecyle d'Angular dans ce cas ci un `ngOnInit` devrait être suffisant.

```suggestion
ngOnInit(): void {
      this.hexColor = this.color;
}
```